gcsio/src/main/java/com/google/cloud/hadoop/gcsio/GoogleCloudStorageGrpcWriteChannel.java

```diff
@@ -252,27 +345,42 @@ private WriteObjectResponse doResumableUpload() throws IOException {
                 resourceId, responseObserver.uploadId));
       }
 
+      long numberOfBufferedRequests = channelOptions.getNumberOfBufferedRequests();
+      long freeBufferThreshold =
+          Math.max(numberOfBufferedRequests - 5, numberOfBufferedRequests / 2);
```

Comment:
what are these magic numbers? Can we create a constant and give a meaningful name?

Reply:
added comments to explain this threshold